### DISCUSSION TITLE
ResGatedGraphConv not as performant as dgl equivalent

### DISCUSSION THREAD

**Discussion by Koutoulakis** [pyg-team/pytorch_geometric#7358] [Q&A] [answered]
TL;DR.  The ResGatedGraphConv in pytorch geometric, is missing the edge features, compared to the dgl equivalent, and tests I've run suggest that its results are not as performant. Could someone help to update the code/create a new model as I have no clue how the MessagePassing  code works?

More explanation.
I've tried using the [ResGatedGraphConv](https://github.com/pyg-team/pytorch_geometric/blob/6092039ebcaa2f769d8f11eb919a1cc2bcfb3db1/torch_geometric/nn/conv/res_gated_graph_conv.py#L12)  which is pretty much equation (12) in https://arxiv.org/pdf/1711.07553v2.pdf 
However, it seems that the author's equations don't contain the edge feature update as can be seen in the equations (16,17,18) from 
 this paper: https://arxiv.org/pdf/2003.00982.pdf which can be found in this dgl [repo](https://github.com/graphdeeplearning/benchmarking-gnns/blob/b6c407712fa576e9699555e1e035d1e327ccae6c/layers/gated_gcn_layer.py#L12)
 The results suggest that the dgl model performs better in my dataset, but I would like to use pytorch geometric as it has better serialization tools and is more shippable.
 Can someone help to update the code/make a different model with the model described in https://arxiv.org/pdf/2003.00982.pdf as I'm kind of clueless? Or am I misinterpreting what I'm reading and I'm using the wrong model?
 
 Thanks yall!!!

**Comment by rusty1s** [comment, marked as answer]
You can find an alternative PyG implementation of this layer in this repository, see [here](https://github.com/karl-zhao/benchmarking-gnns-pyg/blob/master/layers/gated_gcn_layer.py#L95). Hopefully, this unblocks you. I am happy to take add this layer to PyG in. I am happy to help you with making this contribution.

**Reply by Koutoulakis** [reply]
Thanks!!! I've given it a try. It looks like it works much better, for my use-case at least! 
Is there a specific reason the one you've posted doesn't contain the batch-normalization/relu/residual steps? (The one class above that contains them (dgl graph input one*), so I might be missing something here. I've tried adding them, but after the 50th epoch, things got prohibitively slow for some reason).
What is the exact contribution? Moving the layer you've posted under the existing one?(https://pytorch-geometric.readthedocs.io/en/latest/_modules/torch_geometric/nn/conv/res_gated_graph_conv.html)
Under a new one?
I would be more than happy to do it, even if it's just to learn the procedure!
